### PR TITLE
WIP - Update Xamarin Forms Version

### LINKED ISSUE #100: Update Xamarin.Forms dependency specification
Although the package requires the Xamarin.Forms version greater than or equal to _1.5.0.6447_ the version _2.0.0.6490_ does not work. I had to update to version _2.3.2.127_ to run.

Exception:
```
8-31 16:40:09.747	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	Probing 'java_interop_jnienv_throw'.
08-31 16:40:09.747	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	Searching for 'java_interop_jnienv_throw'.
08-31 16:40:09.747	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	Found as 'java_interop_jnienv_throw'.
08-31 16:40:09.747	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	DllImport searching in: '__Internal' ('(null)').
08-31 16:40:09.743	 5" KitKat (4.4) XXHDPI Phone	Warning	1512	dalvikvm	VFY: unable to resolve direct method 17860: Ljava/lang/Error;.<init> (Ljava/lang/String;Ljava/lang/Throwable;ZZ)V
08-31 16:40:09.743	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	Searching for 'java_interop_jnienv_new_string'.
08-31 16:40:09.743	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	dalvikvm	VFY: replacing opcode 0x70 at 0x0000
08-31 16:40:09.743	 5" KitKat (4.4) XXHDPI Phone	Info	1512	dalvikvm	Could not find method java.lang.Error.<init>, referenced from method android.runtime.JavaProxyThrowable.<init>
08-31 16:40:09.743	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	DllImport searching in: '__Internal' ('(null)').
08-31 16:40:09.743	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	Found as 'java_interop_jnienv_new_string'.
08-31 16:40:09.743	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	Probing 'java_interop_jnienv_new_string'.
08-31 16:40:09.735	 5" KitKat (4.4) XXHDPI Phone	Info	1512	MonoDroid	System.TypeLoadException: Could not resolve type with token 01000025 (from typeref, class/assembly NUnit.Runner.Services.TestOptions, nunit.runner.Droid, Version=3.6.1.0, Culture=neutral, PublicKeyToken=null)
  at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x0000f] in <d278c06ad5684d6882c743a94a93ebc2>:0 
  at (wrapper dynamic-method) System.Object:2520acca-ec1f-416d-abc3-4a879a332195 (intptr,intptr,intptr)
08-31 16:40:09.727	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	Searching for '__android_log_print'.
08-31 16:40:09.727	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	DllImport attempting to load: '/system/lib/liblog.so'.
08-31 16:40:09.727	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	Found as '__android_log_print'.
08-31 16:40:09.727	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	Probing '__android_log_print'.
08-31 16:40:09.727	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	DllImport searching in: '/system/lib/liblog.so' ('/system/lib/liblog.so').
08-31 16:40:09.727	 5" KitKat (4.4) XXHDPI Phone	Debug	1512	Mono	DllImport loaded library '/system/lib/liblog.so'.
08-31 16:40:09.727	 5" KitKat (4.4) XXHDPI Phone	Info	1512	MonoDroid	UNHANDLED EXCEPTION:
08-31 16:39:52.223	 5" KitKat (4.4) XXHDPI Phone	Error	460	WindowManager	Starting window AppWindowToken{a9439778 token=Token{a9308420 ActivityRecord{a92f5d08 u0 com.tratorimetro.library.verion.tests.android/md5b51696b2978721a0e0e6281775f51f11.MainActivity t5}}} timed out
```

### PR DESCRIPTION
**Pull request by rprouse**
Fixes #100 

This updates Xamarin.Forms to the latest current version along with NUnit to 3.9. My thinking is that the test framework should be using the newest version of Xamarin even if the tests are running an older version of Forms. @ChrisMaddock can you think of any flaws with that logic?

I am going to knock off a few more Xamarin updates based off of this PR. I will rebase if this needs changing.

I've tested on all three platforms.

### CODE REVIEW

File: nuget/uap10.0/MainPage.xaml.txt.pp [hunk 1]
```diff
@@ -4,7 +4,7 @@
     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
     xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
     xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
-    xmlns:forms="using:Xamarin.Forms.Platform.WinRT"
```
Comment (ChrisMaddock) [inline nuget/uap10.0/MainPage.xaml.txt.pp]: nunit-vs-templayes should be updated in sync with this. (Or an issue filed, at least!)

File: src/runner/nunit.runner.Droid/project.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ﻿{
   "dependencies": {
-    "NUnit": "[3.8.1]",
+    "NUnit": "3.9.0",
```
Comment (ChrisMaddock) [inline src/runner/nunit.runner.Droid/project.json]: This should still be fixed to a particular version - I assume removing it was accidental? (This has happened in many of the project.json's)

File: src/runner/nunit.runner.Droid/nunit.runner.Droid.nuget.props [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
```
Comment (ChrisMaddock) [inline src/runner/nunit.runner.Droid/nunit.runner.Droid.nuget.props]: NuGet recommends that *.nuget.props is gitignored. 😄 

https://docs.microsoft.com/en-us/nuget/consume-packages/packages-and-source-control#omitting-packages-with-git

File: src/runner/nunit.runner.uwp/project.json [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
     "Microsoft.NETCore.UniversalWindowsPlatform": "5.2.2",
-    "NUnit": "[3.8.1]",
+    "NUnit": "3.9.0",
     "PCLStorage": "1.0.2",
-    "Xamarin.Forms": "1.5.0.6447"
+    "Xamarin.Forms": "2.5.0.122203"
   },
   "frameworks": {
-    "uap10.0": {}
+    "uap10.0.10240": {}
```
Comment (ChrisMaddock) [inline src/runner/nunit.runner.uwp/project.json]: Was this autogenerated? Not sure why we'd want to specify a particular UWP version here.

File: src/tests/nunit.runner.tests.uwp/MainPage.xaml.cs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-﻿// Copyright (c) 2015 CNUnit Project
+﻿// Copyright (c) 2018 Rob Prouse
 //
```
Comment (ChrisMaddock) [inline src/tests/nunit.runner.tests.uwp/MainPage.xaml.cs]: Were we not going for "NUnit Project" here? I don't mind which - but the inconsistency is odd. 🙂

**Comment by ChrisMaddock** [comment]
I am also concerned about what's breaking CI - I think that should be investigated before merge.

Was this working towards a release? I don't think I ever got back to do one since https://github.com/nunit/nunit/pull/2378 - meaning the current release still doesn't work out the box with the latest Android. I think it would be good to get a new one out if possible. :-)

File: src/runner/nunit.runner.Droid/project.json (hunk 2 above)
Comment (ChrisMaddock) [reply src/runner/nunit.runner.Droid/project.json]: Actually, where these auto-removed when you upgraded? I wonder if switching to new-csproj would help - `dotnet migrate` has been totally pain-free on the things I've tried it on so far...

File: nuget/uap10.0/MainPage.xaml.txt.pp (hunk 1 above)
Comment (rprouse) [reply nuget/uap10.0/MainPage.xaml.txt.pp]: And with the updated versions for Xamarin.Forms, etc. 👍

File: src/runner/nunit.runner.Droid/project.json (hunk 2 above)
Comment (rprouse) [reply src/runner/nunit.runner.Droid/project.json]: Yes, it was when I upgraded in the UI. I would like to switch these to the new project format and maybe even switch the three platform specific projects to .NET Standard and just distribute that.

File: src/runner/nunit.runner.uwp/project.json (hunk 4 above)
Comment (rprouse) [reply src/runner/nunit.runner.uwp/project.json]: I didn't touch that, it must have got auto-generated. Removing...

File: src/tests/nunit.runner.tests.uwp/MainPage.xaml.cs (hunk 5 above)
Comment (rprouse) [reply src/tests/nunit.runner.tests.uwp/MainPage.xaml.cs]: I was going to switch the entire project over to my name after the discussion with Charlie about Nunit Project not being a real thing. I'll do that as a separate PR, but this error caught my attention and my OCD couldn't leave it 😄

**Comment by rprouse** [comment]
I was thinking of releasing this early March after the 3.10 release of the framework? I'm doing a ton of Xamarin development these days, so I would like to get this project up and running again and also want to figure out how to use this in a CI system, auto-deploying to test cloud or local devices for testing.

**Comment by rprouse** [comment]
I'll look at the CI builds and get them working as part of this. Switching title to WIP.

**Comment by rprouse** [comment]
FYI, I tried for hours last night to use `dotnet migrate` to update the projects. I also tried to update the project formats to a newer format but failed. I'm probably not going to be able to get back to this before I leave on vacation tomorrow.

**Comment by ChrisMaddock** [comment]
Really? I'm surprised, it worked straight out the box for the everything I did. Maybe the Xamarin platforms are less supported. Sorry for the bad advice. 😞

**Comment by rprouse** [comment]
Xamarin was the problem. It converted everything cleanly, but it stripped out all of the Xamarin stuff. I think the best bet might be to generate new projects and copy the files in. That might be a bit much for this PR though, so I may just try to get the cake build working again, close this, then do the additional work in other PRs. It would be really nice to get everything up to date with the latest Xamarin project formats and dependencies.

**Comment by rprouse** [comment]
Abandoning this PR in favour of #106

**Pull request closed without merging**